File: system-test/typescript/src/testkit/dsl/port/myShop/when/when-stage.ts
import type { WhenPlaceOrder } from './steps/when-place-order.js';
import type { WhenCancelOrder } from './steps/when-cancel-order.js';
import type { WhenViewOrder } from './steps/when-view-order.js';
import type { WhenPublishCoupon } from './steps/when-publish-coupon.js';
import type { WhenBrowseCoupons } from './steps/when-browse-coupons.js';

export interface WhenStage {
  placeOrder(): WhenPlaceOrder;
  cancelOrder(): WhenCancelOrder;
  viewOrder(): WhenViewOrder;
  publishCoupon(): WhenPublishCoupon;
  browseCoupons(): WhenBrowseCoupons;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/myshop/when/WhenStage.java
package com.mycompany.myshop.testkit.dsl.port.myshop.when;

import com.mycompany.myshop.testkit.dsl.port.myshop.when.steps.WhenBrowseCoupons;
import com.mycompany.myshop.testkit.dsl.port.myshop.when.steps.WhenCancelOrder;
import com.mycompany.myshop.testkit.dsl.port.myshop.when.steps.WhenPlaceOrder;
import com.mycompany.myshop.testkit.dsl.port.myshop.when.steps.WhenPublishCoupon;
import com.mycompany.myshop.testkit.dsl.port.myshop.when.steps.WhenViewOrder;

public interface WhenStage {
    WhenPlaceOrder placeOrder();

    WhenCancelOrder cancelOrder();

    WhenViewOrder viewOrder();

    WhenPublishCoupon publishCoupon();

    WhenBrowseCoupons browseCoupons();
}
